### PR TITLE
Fix update powershell feature on windows

### LINKED ISSUE #2323: Auto-update on Windows fails on default MSI options when executed from Core-based PSIC
### System Details

System Details Output
```
### VSCode version: 1.39.2 6ab598523be7a800d7f3eb4d92d7ab9a66069390 x64

### VSCode extensions:
akamud.vscode-theme-onelight@2.1.0
ban.spellright@3.0.48
blackmist.LinkCheckMD@0.1.5
cssho.vscode-svgviewer@2.0.0
darkriszty.markdown-table-prettify@2.4.0
DavidAnson.vscode-markdownlint@0.32.0
dbaeumer.vscode-eslint@1.9.1
docsmsft.docs-article-templates@0.2.7
docsmsft.docs-authoring-pack@0.1.3
docsmsft.docs-markdown@0.2.25
docsmsft.docs-metadata@1.0.0
docsmsft.docs-preview@0.3.35
docsmsft.docs-yaml@0.2.4
DotJoshJohnson.xml@2.5.0
DougFinke.vscode-pandoc@0.0.8
eamodio.gitlens@10.2.0
GitHub.vscode-pull-request-github@0.12.0
jock.svg@1.0.5
monokai.theme-monokai-pro-vscode@1.1.14
ms-azuretools.vscode-azurefunctions@0.20.0
ms-vscode.azure-account@0.8.7
ms-vscode.csharp@1.21.7
ms-vscode.powershell@2019.11.0
ms-vscode.powershell-preview@2019.11.0
msazurermtools.azurerm-vscode-tools@0.8.2
PKief.material-icon-theme@3.9.1
redhat.vscode-yaml@0.6.1
streetsidesoftware.code-spell-checker@1.7.20
tht13.html-preview-vscode@0.2.5
vscodevim.vim@1.11.3
yiwwan.vscode-scope@0.0.21
zhuangtongfa.Material-theme@2.28.3


### PSES version: 2.0.0.0

### PowerShell version:

Name                           Value
----                           -----
PSVersion                      7.0.0-preview.6
PSEdition                      Core
GitCommitId                    7.0.0-preview.6
OS                             Microsoft Windows 10.0.19508
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Issue Description

1. Got the update notification for preview.6 while sitting in a preview.5 PSIC. 
1. Confirmed no other `pwsh` was running. 
1. Clicked yes on the update notification.
    * Confirmed that the PSIC was synchronously blocked on execution of the MSI.
1. Clicked through the MSI with default options
1. Got the dialog complaining that a `pwsh` is still open, and so the installer can't proceed.
1. Left the radio box selected that closes all `pwsh` instances and attempts to restart them.
    * As opposed to the one that finishes install on restart.
1. The PSIC `pwsh` gets terminated by the MSI, and the MSI gets terminated because it's a child process of `pwsh`. 
1. When the PSIC restarts, it spews an error about broken pipelines (presumably because the PowerShell install is now in a half-way broken state having been terminated abruptly).

#### Expected Behavior

Clicking Yes -> Next -> Next -> Next etc. should update you to the latest version of PowerShell on Windows. 

Not sure if this problem is happening on other platforms, but this could be fixed on Windows by automatically setting the PSIC to Windows PowerShell 5.1 before proceeding with the install script. (Though I guess this wouldn't work in future versions of the extension where 3.0 and 4.0 aren't supported.....it could drop the PSIC altogether and run the script in a PATH'd powershell.exe terminal instance.)

#### Actual Behavior

The MSI process is ended abruptly, and I'm left with a corrupted, partial update of PowerShell. 

### Attached Logs

I have these handy, but I don't want to scrub them right now, and I think the busted behavior is clear. If you really need 'em, let me know.

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
## PR Summary

fixes #2323 

This uses the node runtime to fetch the MSI and then uses cmd to execute the MSI. This will also stop the PSIC and then restart it when the MSI is done.

## PR Checklist

Note: Tick the boxes below that apply to this pull request by putting an `x` between the square brackets.
Please mark anything not applicable to this PR `NA`.

- [x] PR has a meaningful title
- [x] Summarized changes
- [ ] PR has tests
- [x] This PR is ready to merge and is not work in progress
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready

### CODE REVIEW

File: src/features/UpdatePowerShell.ts [hunk 1]
```diff
@@ -116,39 +126,55 @@ export async function InvokePowerShellUpdateCheck(
...
+                async () => {
+                    // Streams the body of the request to a file.
+                    await streamPipeline(res.body, fs.createWriteStream(msiDownloadPath));
+                });
+
+                // Stop the Integrated Console session because Windows likes to hold on to files.
+                sessionManager.stop();
+
+                // Invoke the MSI via cmd.
+                const msi = spawn("cmd", [`/S /C ${msiDownloadPath}`], {
```
Comment (rjmholt) [inline src/features/UpdatePowerShell.ts]: Do we need to use `cmd`, or is there a more direct way of invoking the MSI (or is there a reason not to)?

File: src/features/UpdatePowerShell.ts [hunk 2]
```diff
@@ -116,39 +126,55 @@ export async function InvokePowerShellUpdateCheck(
...
+
+                msi.on("close", (code) => {
+                    // Now that the MSI is finished, start the Integrated Console session.
+                    sessionManager.start();
+                    fs.unlinkSync(msiDownloadPath);
+                });
 
             } else if (isMacOS) {
-                script = "brew cask upgrade powershell";
+                let script = "brew cask upgrade powershell";
```
Comment (rjmholt) [inline src/features/UpdatePowerShell.ts]: Possible case for a ternary?

File: src/features/UpdatePowerShell.ts [hunk 3]
```diff
@@ -116,39 +126,55 @@ export async function InvokePowerShellUpdateCheck(
...
+                });
+
+                // Stop the Integrated Console session because Windows likes to hold on to files.
+                sessionManager.stop();
+
+                // Invoke the MSI via cmd.
+                const msi = spawn("cmd", [`/S /C ${msiDownloadPath}`], {
+                    detached: true,
+                    cwd: os.homedir(),
+                    env: process.env,
```
Comment (rjmholt) [inline src/features/UpdatePowerShell.ts]: Does spawn not pass the environment by default?

File: src/features/UpdatePowerShell.ts [hunk 4]
```diff
@@ -116,39 +126,55 @@ export async function InvokePowerShellUpdateCheck(
...
+                    await streamPipeline(res.body, fs.createWriteStream(msiDownloadPath));
+                });
+
+                // Stop the Integrated Console session because Windows likes to hold on to files.
+                sessionManager.stop();
+
+                // Invoke the MSI via cmd.
+                const msi = spawn("cmd", [`/S /C ${msiDownloadPath}`], {
+                    detached: true,
+                    cwd: os.homedir(),
```
Comment (rjmholt) [inline src/features/UpdatePowerShell.ts]: Does execution of the MSI depend on this setting?

File: src/features/UpdatePowerShell.ts (hunk 1 above)
Comment (SeeminglyScience) [reply src/features/UpdatePowerShell.ts]: What about `"msiexec", [ "/i", msiDownloadPath ]`

File: src/features/UpdatePowerShell.ts (hunk 4 above)
Comment (TylerLeonhardt) [reply src/features/UpdatePowerShell.ts]: simplified

File: src/features/UpdatePowerShell.ts (hunk 3 above)
Comment (TylerLeonhardt) [reply src/features/UpdatePowerShell.ts]: not needed. You're right

File: src/features/UpdatePowerShell.ts (hunk 2 above)
Comment (TylerLeonhardt) [reply src/features/UpdatePowerShell.ts]: ye

File: src/features/UpdatePowerShell.ts (hunk 1 above)
Comment (TylerLeonhardt) [reply src/features/UpdatePowerShell.ts]: Yes! Changed to msiexec

**Pull request merged**